tensorflow_lite_support/python/task/vision/image_classifier.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Image classifier task."""
+
+import dataclasses
+from typing import Optional
+
+from tensorflow_lite_support.python.task.core import task_options
+from tensorflow_lite_support.python.task.core.proto import configuration_pb2
+from tensorflow_lite_support.python.task.processor import processor_options
+from tensorflow_lite_support.python.task.processor.proto import bounding_box_pb2
+from tensorflow_lite_support.python.task.processor.proto import classifications_pb2
+from tensorflow_lite_support.python.task.processor.proto import image_classifier_options_pb2
+from tensorflow_lite_support.python.task.vision.core import tensor_image
+from tensorflow_lite_support.python.task.vision.core.pybinds import image_utils
+from tensorflow_lite_support.python.task.vision.pybinds import image_classifier as _image_classifier
+
+
+@dataclasses.dataclass
+class ImageClassifierOptions:
+  """Options for the image classifier task."""
+  base_options: task_options.BaseOptions
+  classifier_options: Optional[processor_options.ClassificationOptions] = None
+
+
+def _build_proto_options(
+    options: ImageClassifierOptions
+) -> image_classifier_options_pb2.ClassifierOptions:
+  """Builds the protobuf image classifier options."""
+  # Builds the initial proto_options.
+  proto_options = image_classifier_options_pb2.ClassifierOptions()
+
+  # Updates values from base_options.
+  proto_options.model_file_with_metadata.file_name = options.base_options.model_file
+  proto_options.num_threads = options.base_options.num_threads
+  if options.base_options.use_coral:
+    proto_options.compute_settings.tflite_settings.delegate = configuration_pb2.Delegate.EDGETPU_CORAL
+
+  # Updates values from classifier_options.
```

Comment:
As these dataclass to proto conversion logic will be reused in other tasks, please put them on a shared module (e.g `python/task/core/utils.py`) for better code reuse.